x-pack/plugins/apm/public/components/shared/charts/spark_plot/index.tsx

```diff
@@ -33,19 +35,29 @@ type Color =
   | 'euiColorVis8'
   | 'euiColorVis9';
 
+function hasValidTimeseries(
+  series?: Coordinate[] | null
+): series is Coordinate[] {
+  return !!series?.some((point) => point.y !== null);
```

Comment:
It took me a second look to understand what this was checking, wondering if we should make it easier to read, at the cost of searching the whole array?
```suggestion
  return !series?.every(({ y }) => y === null);
```

Reply:
oh nevermind, i see further down that this was an intentional change.

Reply:
if I check with only one `!` and the series is undefined it returns true when it should return false, that's why the second one is necessary.

I can do:

```js
return series && series.some((point) => point.y !== null);
```
WDYT?